### ISSUE TITLE
DMS 9.0 sync throws The type initializer for 'Microsoft.Data.Sqlite.SqliteConnection' threw an exception.

### ISSUE THREAD

**Issue by VagueGit** [Mimetis/Dotmim.Sync#632]
Was using `DMS 0.8.1-beta-0687`. Only change is upgrade client and web api to `DMS 0.9.0`. Also upgraded `Microsoft.Data.Sqlite 5.0.8` to` 6.0.0` as required by DMS

Stack trace
```
   at Dotmim.Sync.SyncAgent.<>c__DisplayClass53_0.<<SynchronizeAsync>b__0>d.MoveNext()
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Dotmim.Sync.SyncAgent.<SynchronizeAsync>d__53.MoveNext()
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
   at Wbl.Services.DataService.<SyncAsync>d__46.MoveNext() in C:\Repos\FREE\Wbl\Services\DataService.cs:line 410
```
Inner exception stack trace
```
   at Microsoft.Data.Sqlite.SqliteConnection..ctor(String connectionString)
   at Dotmim.Sync.Sqlite.SqliteSyncProvider.CreateConnection()
   at Dotmim.Sync.BaseOrchestrator.<RunInTransactionAsync>d__67`1.MoveNext()
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Dotmim.Sync.SyncAgent.<>c__DisplayClass53_0.<<SynchronizeAsync>b__0>d.MoveNext()
```
Rolled back to `DMS 0.8.1-beta-0687` and all is tickety-boo

**Comment by vunder** [comment]
Having the same issue.
Looks like this is because of upgrade to .Net6. App project is not .net6 yet, so trying to load libraries for sqlite fails.
Solution - try to explicitly set version of sqlite library for app project (I mean `assemblyBinding`). I have not tried this yet, may not be the case

**Comment by Mimetis** [comment]
Thanks. Will look into it.
Thats strange because all the tests are green....

**Comment by vunder** [comment]
I'll try to prepare some sample, but the common case - create platform-specific app (e.g. UWP app) with Dotmim 0.8.0 and the update packages to 0.9.0
**Important note**. This is happening not within Dotmim library, but when application itself it using `Microsot.Data.Sqlite` package. E.g. creating `SqliteConnectionStringBuilder`

**Comment by Mimetis** [comment]
Thanks @vunder and @VagueGit .

Can you try with the last version `v0.9.1` ?

**Comment by vunder** [comment]
@Mimetis Did not work for me
I've created a sample app. https://1drv.ms/u/s!Aky6OedVga19k9JzMg8ZVg1yKEIOrQ?e=XSXiiw
First zip with 0.8.0 with Sqlite 5.0.3
Second - updated to 0.9.1 with Sqlite 6.0.0

**Comment by vunder** [comment]
@Mimetis No, I was wrong. `0.9.1` works good in case keeping project Sqlite version 5.0.3. Thanks for quick fix

**Comment by VagueGit** [comment]
I tested `DMS 0.9.1` like this:
Install on test server only with
```
bool upgrade = await orchestrator.NeedsToUpgradeAsync();
if (upgrade)
    await orchestrator.UpgradeAsync();
```
All good there. Then I applied `DMS 0.9.1` to the client which has `SQLite 5.0.12`. All good there.

Then I upgraded the client to `SQLite 6.0.0 `and the exceptions threw again. I note SQLite 6.0.0 requires `.NETStandard 2.0`. Should this be closed if we can only use `DMS 0.9.1` with `SQLite<6.0.0`?

**Comment by vunder** [comment]
@VagueGit I'd say it depends on your project. In my case, host app is UWP, so I cannot target `Sqlite 6.0.0` and had to keep `5.0.3`

**Comment by VagueGit** [comment]
I don't understand @vunder . UWP supports `.NETStandard2.0` and `SQLite 6.0.0` supports `.NetStandard2.0`. Doesn't that mean you should be able to use `SQLite 6.0.0` in a UWP app?

We use SQLite in a `.NetStandard2.0` library. SQLite is in widespread use and devs would expect to be able to use the latest version. If DMS crashes the app when the app uses `SQLite 6.0.0` then that will be an issue for users of DMS.

**Comment by vunder** [comment]
@VagueGit You're right about theoretical compatibility. I did not deep dive into this issue. I assume there are some other dependencies that somehow conflicts with UWP. It should be "based on .Net Core" but what I've discovered with my experience is that it depends on your minimum/target version and version of Microsoft.NETCore.UniversalWindowsPlatform package version

**Comment by Mimetis** [comment]
@VagueGit I don't know what the problem is with `Sqlite`, but it seems that the version `6.0.0` does not work well if you are not targeting a `.net 6.0 `application

**Comment by VagueGit** [comment]
My apologies @Mimetis and @vunder . You are correct. I tested SQLite 6.0.0 in a .NETStandard 2.0 project without sync and it threw the same type of exception. I guess this is one for @bricelam to figure out

**Comment by Mimetis** [comment]
Indeed, that's why I reworked the version v0.9.1 to distinguish the sqlite version, depending on the framework used:

https://github.com/Mimetis/Dotmim.Sync/blob/d604dbe3e45bf4c0826d8003096857f60796c273/Projects/Dotmim.Sync.Sqlite/Dotmim.Sync.Sqlite.csproj#L48-L58

**Comment by bricelam** [comment]
This should be fixed in version 6.0.1 of Microsoft.Data.Sqlite (due any day now)
